### PR TITLE
test(validation): detect ghost package directories

### LINKED ISSUE #1448: [audit][S1 Project Structure] Ghost test directory: tests/unit/git/ has no source files
**Audit finding** — strict full-coverage audit (2026-06-16)

**Section:** 1 — Project Structure
**Severity:** MAJOR
**Location:** `tests/unit/git/`
**Principle:** MODULARITY

## Detail

Same deletion in commit 76b46c9 removed tests/unit/git/__init__.py, tests/unit/git/test_changelog.py, and tests/unit/git/test_git_utils.py. The directory remains with only __pycache__ containing stale .pyc files. The mirror check (hephaestus-check-test-structure) treats this as a valid mirror for hephaestus/git/, so both ghost directories together create a false-positive passing check rather than exposing that neither the source package nor its tests have any content.

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Add detection for ghost test directories with no backing source files or empty source packages, preventing false-positive validation checks.

## Changes
- Add hephaestus_check_ghost_packages validator to detect test dirs with no corresponding source content
- Add hephaestus_check_ghost_test_dirs validator to detect empty test mirrors for non-existent or empty source packages
- Extend test_structure.py validation suite with ghost package detection logic
- Add comprehensive test coverage in test_test_structure.py for both ghost validators

## Testing
- Unit tests verify ghost test directory detection (tests/unit/git/ scenario)
- Unit tests verify ghost package detection (empty source packages)
- Validation suite tests confirm validators reject ghost directories while passing valid structures

## Closes
Closes #1448

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: hephaestus/validation/test_structure.py [hunk 1]
```diff
@@ -74,6 +78,51 @@ def _get_subpackages(root: Path) -> set[str]:
...
+    Args:
+        src_root: Path to the source package (e.g. ``hephaestus/``).
+        test_root: Path to the unit test root (e.g. ``tests/unit/``).
+
+    Returns:
+        Tuple of ``(ok, ghosts)`` where *ghosts* is the set of subpackage names
+        whose source dir has no module AND whose test dir has no ``test_*.py``.
+
+    """
+    shared = _get_subpackages(src_root) & _get_subpackages(test_root)
```
Comment (mvillmow) [inline hephaestus/validation/test_structure.py:117]: Scope note (no fix required): `check_no_ghost_packages` only considers `_get_subpackages(src_root) & _get_subpackages(test_root)`, and `_get_subpackages` excludes dirs lacking any Python source (the `__pycache__`-only case `_has_python_source` filters out). So the issue's *literal* described state — a `tests/unit/git/` containing only stale `.pyc` files — is not enumerated and thus not flagged here. That's correct scoping rather than a gap: a `__pycache__`-only dir is not a subpackage on either side, so the name-only mirror check never treats it as a 'valid mirror' to begin with. This check correctly targets the actual false-positive surface (an `__init__.py`-only shell pair). Consider a one-line comment here recording that the content-free pair must carry `__init__.py` to be in scope, so a future reader doesn't expect `__pycache__`-only detection.

**Review by mvillmow — COMMENTED** [review]
GO — additive ghost-pair check correctly closes the name-only-mirror false positive; helper + orchestrator tests; ships green on real tree.

**Review by mvillmow — COMMENTED** [review]
LGTM — content-aware ghost-package check correctly closes the name-only mirror false-positive; both-altitude tests, ships green, no new CI job.
